llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -1224,6 +1238,22 @@ static bool doesNotNeedToSchedule(ArrayRef<Value *> VL) {
          (all_of(VL, isUsedOutsideBlock) || all_of(VL, areAllOperandsNonInsts));
 }
 
+/// Returns true if widened type of \p Ty elements with size \p Sz represents
+/// full vector type, i.e. adding extra element results in extra parts upon type
+/// legalization.
+static bool hasFullVectorsOnly(const TargetTransformInfo &TTI, Type *Ty,
+                               unsigned Sz) {
+  if (Sz <= 1)
+    return false;
+  if (!isValidElementType(Ty) && !isa<FixedVectorType>(Ty))
```

Comment:
I'm not certain but should this be: `!isValidElementType(Ty) || !isa<FixedVectorType>(Ty)`?

Reply:
No, it is correct, fixed vector type is acceptable (for insertelement instruction and revec)